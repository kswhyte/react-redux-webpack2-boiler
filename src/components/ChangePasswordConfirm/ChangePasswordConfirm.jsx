import React from 'react';
import { Link } from 'react-router-dom';

import './change-password-confirm.css';

const ChangePasswordConfirm = () => {
  return (
    <div className="change-password-confirm-wrapper card">
      <img src="images/pattern-dots.png" className="dots-pattern" />
      <h2>PASSWORD UPDATED</h2>
      <div className="page-instructions md-spacer col-xs-offset-3 col-xs-6">
        <p>
          Your password has been updated. Follow the link below to login with your updated credentials.
        </p>
      </div>

      <div className="row lg-spacer">
        <div className="col-xs-offset-3 col-xs-6 sm-spacer">
          <span className="lnr lnr-checkmark-circle" />
        </div>
      </div>

      <form>
        <div className="row sm-spacer">
          <div className="col-xs-offset-3 col-xs-6 sm-spacer">
            <Link to="/login" className="primary-btn btn-link">
              SIGN IN
            </Link>
          </div>
        </div>
      </form>
    </div>
  );
};

ChangePasswordConfirm.propTypes = {};

export default ChangePasswordConfirm;
