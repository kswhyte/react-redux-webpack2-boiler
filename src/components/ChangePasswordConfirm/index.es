import Component from './ChangePasswordConfirm';
export default Component;
