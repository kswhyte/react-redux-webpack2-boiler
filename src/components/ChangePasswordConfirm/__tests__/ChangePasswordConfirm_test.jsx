import React from 'react';
// import { storiesOf, action, linkTo } from '@kadira/storybook';
import ChangePasswordConfirm from '../ChangePasswordConfirm';

// import { specs, describe, it } from 'storybook-addon-specifications';

import { storiesOf, describe, it, specs } from '../../../../.storybook/facade';

import { mount } from 'enzyme';
import { StaticRouter } from 'react-router';
import expect from 'expect';

const stories = storiesOf('ChangePasswordConfirm', module);
const props = {
}
stories.add('ChangePasswordConfirm Page Story', () => {
  const changePasswordConfirmStory = (
    <StaticRouter {...props} context={{}}>
      <ChangePasswordConfirm {...props} />
    </StaticRouter>
  );
  specs(() =>
    describe('Show a successful alert', () => {
      it('Should render the ChangePasswordConfirm component without crashing', () => {
        let output = mount(changePasswordConfirmStory);
        expect(output.find('.change-password-confirm-wrapper').exists()).toBe(true)
      });
      it('Should have a header element that says "Password Updated"', () => {
        let output = mount(changePasswordConfirmStory);
        expect(output.find('h2').text()).toContain('PASSWORD UPDATED');
      });
      it('Should have correct p tag contents', () => {
        let output = mount(changePasswordConfirmStory);
        expect(output.find('p').text()).toContain('Your password has been updated. Follow the link below to login with your updated credentials.');
      });

    })
  );
  return changePasswordConfirmStory;
});
