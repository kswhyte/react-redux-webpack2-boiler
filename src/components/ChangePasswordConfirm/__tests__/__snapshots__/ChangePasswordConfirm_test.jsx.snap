// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePasswordConfirm Page Story 1`] = `
<div
  className="change-password-confirm-wrapper card"
>
  <img
    className="dots-pattern"
    src="images/pattern-dots.png"
  />
  <h2>
    PASSWORD UPDATED
  </h2>
  <div
    className="page-instructions md-spacer col-xs-offset-3 col-xs-6"
  >
    <p>
      Your password has been updated. Follow the link below to login with your updated credentials.
    </p>
  </div>
  <div
    className="row lg-spacer"
  >
    <div
      className="col-xs-offset-3 col-xs-6 sm-spacer"
    >
      <span
        className="lnr lnr-checkmark-circle"
      />
    </div>
  </div>
  <form>
    <div
      className="row sm-spacer"
    >
      <div
        className="col-xs-offset-3 col-xs-6 sm-spacer"
      >
        <a
          className="primary-btn btn-link"
          href="/login"
          onClick={[Function]}
        >
          SIGN IN
        </a>
      </div>
    </div>
  </form>
</div>
`;
